src/transformers/models/mixtral/modeling_mixtral.py

```diff
@@ -868,11 +811,18 @@ def forward(self, hidden_states: torch.Tensor) -> torch.Tensor:
             expert_layer = self.experts[expert_idx]
             idx, top_x = torch.where(expert_mask[expert_idx])
 
+            if top_x.shape[0] == 0:
+                continue
+
+            # in torch it is faster to index using lists than torch tensors
+            top_x_list = top_x.tolist()
+            idx_list = idx.tolist()
+
```

Comment:
😢 did this also help? We removed it to support #29858. Would also be supprised that this is causing issues. Could you detail what is going on here?

Reply:
I don't think it's this, as I mentioned, most probably related to the rotary embeddings related code which was inherited all the way back [from GPT-Neo](https://twitter.com/rafacelente/status/1787229662533357922)